app/routes/geostore/geostore.py

```diff
@@ -1,47 +1,67 @@
-"""Retrieve a geometry using its mb5 hash for a given dataset, user defined
+"""Retrieve a geometry using its md5 hash for a given dataset, user defined
 geometries in the datastore."""
 
 from uuid import UUID
 
 from fastapi import APIRouter, Depends, Path
 from fastapi.responses import ORJSONResponse
 
+from ...crud import geostore
+from ...models.pydantic.geostore import GeostoreHydrated, GeostoreIn, GeostoreResponse
 from ...routes import dataset_dependency, version_dependency
 
 router = APIRouter()
 
 
 @router.post(
-    "/geostore", response_class=ORJSONResponse,
+    "/geostore",
+    response_class=ORJSONResponse,
+    response_model=GeostoreResponse,
+    status_code=201,
 )
-async def add_new_geostore():
-    """Add geostore feature to User geostore."""
-    pass
+async def add_new_geostore(
+    *, request: GeostoreIn, response: ORJSONResponse,
+):
+    """Add geostore feature to user area of geostore."""
+
+    input_data = request.dict(exclude_none=True, by_alias=True)
+
+    new_user_area: GeostoreHydrated = await geostore.create_user_area(**input_data)
```

Comment:
Need to catch `BadRequestError` and make it an `HTTPException` with status code 400

Reply:
Done, thank you.